test/unit/services/awsAdapter_spec.js

```diff
@@ -148,4 +150,88 @@ describe("The AwsAdapter class ", function () {
 			});
 		});
 	});
-});
+
+	describe("gets properties of given instance", function () {
+		var describeInstancesStub;
+		var result;
+		before(function () {
+			var awsAdapter = new AwsAdapter(ec2);
+			describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+				var data = { Reservations : [ { Instances : [ { PublicIpAddress : "127.0.0.1" } ] } ] };
+				return Bluebird.resolve(data);
+			});
+			awsAdapter.describeInstance(VALID_EC2_INSTANCE)
+			.then(function (response) {
+				result = response;
+			})
+			.catch(function (error) {
+				result = error;
+			});
+		});
+
+		after(function () {
+			describeInstancesStub.restore();
+		});
+
+		it("gets properties of the given instance", function () {
+			expect(result).to.be.instanceof(Array);
+			expect(result[ 0 ].PublicIpAddress).to.be.equal("127.0.0.1");
+		});
+	});
+
+	describe("gets properties of invalid instance", function () {
+		var describeInstancesStub;
+		var result;
+		before(function () {
+			var awsAdapter = new AwsAdapter(ec2);
+			describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+				return Bluebird.reject(new Error("Instance not Found"));
+			});
+			awsAdapter.describeInstance(VALID_EC2_INSTANCE)
+			.then(function (response) {
+				result = response;
+			})
+			.catch(function (error) {
+				result = error;
+			});
+		});
+
+		after(function () {
+			describeInstancesStub.restore();
+		});
+
+		it("fails", function () {
+			expect(result,"error").to.be.an.instanceof(Error);
```

Comment:
Is it worth inspecting the attributes of the error to make sure that it is the error object that we expect?